### ISSUE TITLE
[Bug]: initdb doesn't wait for secret

### ISSUE THREAD

**Issue by evilhamsterman** [cloudnative-pg/cloudnative-pg#6069]
### Is there an existing issue already for this bug?

- [x] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [x] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [x] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

dmills@qumulo.com

### Version

1.24.0

### What version of Kubernetes are you using?

1.30

### What is your Kubernetes environment?

Self-managed: k3s

### How did you install the operator?

Helm

### What happened?

The initdb option to create a new DB doesn't appear to wait for the secret to exist before starting. Then later if the secret is created the owner is not created unless you separately use a role.

We use external-secrets to save secrets in a secure vault, but if we deploy the external-secret and the cluster at the same time they both start working even though external-secrets may have not created the Kubernetes secret. So you end up with a DB that doesn't have a owner can the user cannot connect. If you recreate the cluster once the secret exists everything proceeds normally

### Cluster resource

```shell
---
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name: db
  namespace: dv
spec:
  instances: 2
  storage:
    size: 10Gi
  bootstrap:
    initdb:
      database: db
      owner: db
      secret:
        name: db-secret

---
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: db-secret
  namespace: db
spec:
  refreshInterval: 1h
  secretStoreRef:
    name: secretstore
    kind: ClusterSecretStore
  target:
    name: db-secret
    template:
      type: kubernetes.io/basic-auth
      data:
        username: "{{ .login }}"
        password: "{{ .password }}"
        host: "{{ .host }}"
  dataFrom:
    - extract:
        key: db-secret
```

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by armru** [comment]
It would be helpful to have the full logs of both the operator and the primary instance (while there is no 'owner')

**Comment by firstdorsal** [comment]
I'm facing the same issue. Currently, the secret is just silently ignored if it's not present right away, a behavior similar to Kubernetes itself when it is missing a secret to mount, (Stuck in ContainerCreating until the secret is available) would be good IMO (Wait and retry until the secret is available).

**Comment by evilhamsterman** [comment]
Still appears to be an issue and PR #6668 looks like it should fix it

**Comment by evilhamsterman** [comment]
Still there

**Comment by evilhamsterman** [comment]
I think it's still there, if it's on the backlog this shouldn't get marked stale

**Comment by kamikredstone** [comment]
I used sync waves in ArgoCD as a workaround, works well. Annotate the Cluster with a higher sync-wave than an ExternalSecret, and add this as a customized health check for ExternalSecrets object in ArgoCD chart:
```yaml
configs:
  cm:
    resource.customizations.health.external-secrets.io_ExternalSecret: |
      hs = {}
      hs.status = "Progressing"
      hs.message = "Waiting for ExternalSecret to be Ready"
      if obj.status and obj.status.conditions then
        for _, c in ipairs(obj.status.conditions) do
          if c.type == "Ready" and c.status == "True" then
            hs.status = "Healthy"
            hs.message = "ExternalSecret Ready"
          end
          if c.type == "Failed" and c.status == "True" then
            hs.status = "Degraded"
            hs.message = c.message or "ExternalSecret Failed"
          end
        end
      end
      return hs
```

**Comment by dominik-niebuhr** [comment]
@armru Will the fix also work for secrets of managed roles? If not should I open a new issue or do we keep the existing issue open for that?
